network/src/node.rs

```diff
@@ -60,12 +60,15 @@ pub struct InnerNode {
     pub config: Config,
     /// The inbound handler of this node.
     pub inbound: Inbound,
+    /// The cache of node's inbound messages.
+    pub inbound_cache: Mutex<Cache>,
```

Comment:
Should this be a RwLock? seems like it would be better here.

Reply:
I thought about this, but every cache lookup has the potential to perform a write to it (which is very likely to happen), which would require a second, mutable lock operation; FWIW, I detected no performance degradation related to the introduction of this lock.